res/controllers/Denon-MC7000-scripts.js

```diff
@@ -837,6 +837,20 @@ MC7000.censor = function(channel, control, value, status, group) {
     }
 };
 
+// Auto-Loop Button
+MC7000.autoLoop = function(channel, control, value, status, group) {
+    // TODO: handle shift button
+    if (value) { // Button Down == 0x7F
+        if (engine.getValue(group, "loop_enabled")) {
+            engine.setValue(group, "loop_enabled", false);
+            engine.setValue(group, "reloop_toggle", true);
+        } else {
+            engine.setValue(group, "beatloop_activate", true);
+            engine.setValue(group, "beatloop_activate", false);
+        }
+    }
+};
+
```

Comment:
This looks a bit strange to me but I cannot test it as I built up another gear right now...

Shifted buttons do exactly what is written on the controller -> `Reloop`
No need to take care about.

Reply:
You're right, the shift button is working as intended.

And yes, it looks strange, but on Saturday it worked more than 6 hours without issues :smile:

Reply:
> And yes, it looks strange, but on Saturday it worked more than 6 hours without issues :smile:
> 

I could try it and indeed it works great. Cheers!